python/pyarrow/io.pxi

```diff
@@ -851,6 +879,16 @@ cdef class MemoryMappedFile(NativeFile):
     A stream that represents a memory-mapped file.
 
     Supports 'r', 'r+', 'w' modes.
+
+    Examples
+    --------
+    Create a MemoryMappedFile:
```

Comment:
Maybe clarify here that this is creating a _new_ file, and not creating a MemoryMappedFile object that is viewing an existing file? (if my understanding about that is correct)

Reply:
And actually, showing `pa.memory_map` to open an existing file with mmap might be more useful to show here.

Reply:
Yeah, makes perfect sense. Will do.